WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -661,17 +743,55 @@ public void onPostUploaded(OnPostUploaded event) {
         }
     }
 
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPostsSearched(OnPostsSearched event) {
+        mIsSearching = false;
+        hideLoadMoreProgress();
+
+        if (event.isError()) {
+            AppLog.w(AppLog.T.POSTS, "Error searching posts: " + event.error.message);
+            ToastUtils.showToast(getActivity(), "Error searching posts: " + event.error.type);
+        } else {
+            if (event.searchResults != null && !ListUtils.isEmpty(event.searchResults.getPosts())) {
+                hideEmptyView();
+                mSearchResults.addAll(event.searchResults.getPosts());
+                setLocalPostIdsOnSearchResults();
+                getPostListAdapter().setPostList(mSearchResults);
```

Comment:
I think this might be the cause of some of the issues we have had while testing. The thing is, we don't check the state of the UI when we get the callback from `onPostsSearched`. We don't know if the user is still searching or closed the search and working with the regular post list. We need to be very careful when we are working with async flows like this and make sure the callback is handled gracefully. This is especially important for us since the network calls can take a while on mobile.

I haven't given much thought about this yet, but we need to manage the state somehow. If the states we are managing are not complicated, the current flags might be enough, or we might need to add more or check the state of the UI components.

Hope this makes sense!